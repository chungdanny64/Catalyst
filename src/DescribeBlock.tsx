import React from 'react'


export const DescribeBlock: React.FC = () =>{



  // this should have It statement componenets rendered inside
  return (
    <div>
      hey
    </div>
  )
}