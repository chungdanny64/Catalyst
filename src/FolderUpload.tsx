import * as React from 'react'
import * as electron from 'electron'
import {remote} from 'electron'
import * as electronFs from 'fs'


// allows electron to use node module fs 

class FileTree {
    filepath: string
    name: string
    children: FileTree[]


    constructor(filepath: string, name: string){
        this.filepath = filepath
        this.name = name
        this.children = []

    }

    // directory will be root directory path
    createTree(directory: string): FileTree[]{
        let treeElements: FileTree[] = []

        // readdirSync will return all the names of the files in the directory as an array 
        electronFs.readdirSync(directory).forEach((fileName: string) =>{

            if(fileName !== ".git" && fileName !== "node_modules" && fileName !== "dist"){
                let fileInfo = new FileTree(directory + "/" + fileName, fileName)

                // statsync will return an object with properties of the filepath indicated
                let moreFileInfo = electronFs.statSync(fileInfo.filepath)
                
                // if the file that is currently at is a directory it will append the children to it
                if(moreFileInfo.isDirectory()){
                    fileInfo.children = fileInfo.createTree(fileInfo.filepath)
                }

                treeElements.push(fileInfo)
            }

        })
        return treeElements
    }

}

export const FolderUpload: React.FC = () => {

    const dialog = remote.dialog
    async function uploadFolder(){
        // allows users to upload a folder 
        const Project = await dialog.showOpenDialog(
            {properties: ['openDirectory'],
            // the types of files that will be displayed or selected 
            filters: [
                { name: 'Javascript Files', extensions: ['js', 'jsx']},
                { name: 'Typescript Files', extensions: ['ts', 'tsx']},
                { name: 'HTML Files', extensions: ['html']}
            ],
            message: 'please choose a project'
        })

        console.log(Project)
        // if the user cancels the action then undefined will be returned
        // if the user successfully completes the action then a string array will be returned
         // Project is an object that holds canceled (boolean to check if it was cancelled) and filePaths (array of filepaths)
        if(Project && !Project.canceled){
            // holds the directory of the project that was selected 
            let projectDirectory = Project.filePaths[0]

            // use regex to find all \ in the case of a windows user and replace with /
            projectDirectory = projectDirectory.replace(/\\/g, '/')

            // will create a new FileTree object for the root directory
            const rootTree = new FileTree(projectDirectory, "root")

            // will return an array of FileTree objects along with any children associated with it 
            const fileTree = rootTree.createTree(projectDirectory)
            console.log(fileTree)
        }
    }


    return(
        <div>
            <button onClick = {uploadFolder}>Upload Project</button>
        </div>
    )
}
