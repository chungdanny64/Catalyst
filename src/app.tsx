import React from 'react';
import ReactDom from 'react-dom';
import {FolderUpload} from './FolderUpload'



export const App = () => {
  return (
    <div >
      <h1>
        Hello from React World
      </h1>
      <FolderUpload />
    </div>
    
  )
};


