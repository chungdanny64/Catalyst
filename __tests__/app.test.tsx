import { shallow, mount, render } from 'enzyme';
import * as React from 'react';
import { App } from '../src/app';

const wrapper = shallow(<App />);


describe('Hello, Enzyme!', () => {
  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  })
});