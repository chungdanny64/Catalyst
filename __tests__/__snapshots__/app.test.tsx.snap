// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello, Enzyme! renders 1`] = `
<div>
  <h1>
    Hello from React World
  </h1>
  <Component />
</div>
`;
